### ISSUE TITLE
no-array-callback-reference false positives

### ISSUE THREAD

**Issue by carakessler** [sindresorhus/eslint-plugin-unicorn#1193]
I have found a problem with `no-array-callback-reference`:
 
The rule is erroring on a custom implementation of a method called `find` 
We have a class that contains a function `find`, and I am getting an error on passing in a parameter to that function, which is NOT the native array `find`  
<img width="940" alt="Screen Shot 2021-04-19 at 4 33 33 PM" src="https://user-images.githubusercontent.com/3287102/115315826-fd958680-a12c-11eb-8901-52e7f1a9a2f8.png">

Additionally, the parameter `taskName` (as shown in the screenshot below) is not a function, and is in fact a string (although again, it shouldn't matter because this isn't a native array method anyway)
<img width="945" alt="Screen Shot 2021-04-19 at 4 35 41 PM" src="https://user-images.githubusercontent.com/3287102/115315987-49e0c680-a12d-11eb-9fd9-b30616094f86.png">

**Comment by sindresorhus** [comment]
> which is NOT the native array find

There's no way to statically infer that unless the method definition is in the same scope.

**Comment by carakessler** [comment]
Ah that makes sense, I wonder if it makes sense to expose a more ergonomic way to override the rule.. I'll think on that. Thanks for the prompt response!

**Comment by matthiasg** [comment]
We also get this false positive when it should be statically clear that the data passed in is not a callback:

```
const query = {}
const docs = m.find(query) // FAILS here (m is NOT an array etc)
```

**Comment by jase88** [comment]
I also ran into this false positive with Angular using `ng-mocks` library

```typescript
import { ngMocks } from 'ng-mocks';
import { MyComponent } from './my-component';

ngMocks.find(MyComponent)
```

**Comment by Samuel-Therrien-Beslogic** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1480#issue-967623165

> mongoose Models have a `find` function to query a MongoDB Database. I think this could be added to `ignoredCallee`
> 
> ```js
> import { model, Schema, Types } from 'mongoose'
> 
> type Account = { _id: string, id: string }
> const schema = new Schema<Account, Model<Account>, Account>()
> const AccountModel = model('Account', schema)
> 
> AccountModel.find({_id: '000000000000000000000000'})
> ```
> ![image](https://user-images.githubusercontent.com/84519004/129114730-2d9092ab-42ef-4d57-a724-ce2c5f119fb7.png)
